atomic_reactor/util.py

```diff
@@ -667,22 +668,38 @@ def query_registry(image, registry, digest=None, insecure=False, dockercfg_path=
         if username and password:
             auth = requests.auth.HTTPBasicAuth(username, password)
 
+    # In the insecure case, if the registry is just a hostname:port, we don't
+    # know whether to talk HTTPS or HTTP to it, so try both ways
     if not re.match('http(s)?://', registry):
-        registry = 'https://{}'.format(registry)
+        if insecure:
+            registries = ('https://{}'.format(registry), 'http://{}'.format(registry))
+        else:
+            registries = ('https://{}'.format(registry),)
+    else:
+        registries = (registry,)
 
     context = '/'.join([x for x in [image.namespace, image.repo] if x])
     reference = digest or image.tag or 'latest'
     object_type = 'manifests'
     if is_blob:
         object_type = 'blobs'
-    url = '{}/v2/{}/{}/{}'.format(registry, context, object_type, reference)
-    logger.debug("url: {}".format(url))
 
     headers = {'Accept': (get_manifest_media_type(version))}
     kwargs = {'verify': not insecure, 'headers': headers, 'auth': auth}
 
-    response = requests.get(url, **kwargs)
-    response.raise_for_status()
+    for r in registries:
+        url = '{}/v2/{}/{}/{}'.format(r, context, object_type, reference)
+        logger.debug("url: {}".format(url))
+
+        try_again = False
+        try:
```

Comment:
Yeah. If I do a fixup patch for that, is it preferred style to provide it as separate commit, or to push a new version of the patches with the fixes squashed?